### PR TITLE
FEC-4042 FairPlay support.

### PR DESCRIPTION
**Pull request by noamtamim**
Refactored parts of KPlayerFactory, created KPAssetBuilder.

### CODE REVIEW

File: KALTURAPlayerSDK/KPAssetBuilder.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@end
+
+
+// Protocol used by KPAssetBuilder to handle DRM operations for an asset.
+
+@protocol KPAssetHandler <NSObject>
+
+-(instancetype)initWithAssetReadyCallback:(KPAssetReadyCallback)callback;
+-(void)setContentUrl:(NSString*)url;
+-(void)setLicenseUri:(NSString *)licenseUri;
```
Comment (nissopa) [inline KALTURAPlayerSDK/KPAssetBuilder.h]: Don't like the name Uri

File: KALTURAPlayerSDK/KPAssetBuilder.m [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
++(NSData*)getCertificate {
+    return s_certificate;
+}
+
+-(instancetype)initWithReadyCallback:(KPAssetReadyCallback)callback {
+    self = [super init];
+    if (self) {
+        _assetReadyCallback = callback;
+    }
```
Comment (nissopa) [inline KALTURAPlayerSDK/KPAssetBuilder.m]: You have to copy the block

File: KALTURAPlayerSDK/KPAssetBuilder.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+@end
+
+
+// Protocol used by KPAssetBuilder to handle DRM operations for an asset.
+
+@protocol KPAssetHandler <NSObject>
+
+-(instancetype)initWithAssetReadyCallback:(KPAssetReadyCallback)callback;
+-(void)setContentUrl:(NSString*)url;
```
Comment (nissopa) [inline KALTURAPlayerSDK/KPAssetBuilder.h]: Why not using property?

File: KALTURAPlayerSDK/KPAssetBuilder.m [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+#import "KPAssetBuilder.h"
+
+#import "KPFairPlayHandler.h"
+#import "KPWidevineClassicHandler.h"
+
+#import "KPLog.h"
+
+@interface KPAssetBuilder () {
+    KPAssetReadyCallback _assetReadyCallback;
+    id<KPAssetHandler> _assetHandler;
```
Comment (nissopa) [inline KALTURAPlayerSDK/KPAssetBuilder.m]: why not using property with (nonatomic, copy)

File: KALTURAPlayerSDK/KPAssetBuilder.m [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        KPLogError(@"Failed parsing content url, can't continue");
+        return;            
+    }
+    
+    Class handlerClass;    
+    if ([contentUrl.pathExtension.lowercaseString isEqualToString:@"wvm"]) {
+        handlerClass = [KPWidevineClassicHandler class];
+    } else {
+        handlerClass = [KPFairPlayHandler class];
+    }
```
Comment (nissopa) [inline KALTURAPlayerSDK/KPAssetBuilder.m]: Why not using Category for string which will return Class

File: KALTURAPlayerSDK/KPFairPlayHandler.m [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+#import "KPFairPlayHandler.h"
+#import "KPAssetBuilder.h"
+#import "KPLog.h"
+
+NSString* const URL_SCHEME_NAME = @"skd";
+
+@interface KPFairPlayHandler () {
+    NSString* _licenseUri;
+    KPAssetReadyCallback _assetReadyCallback;
```
Comment (nissopa) [inline KALTURAPlayerSDK/KPFairPlayHandler.m]: properties with copy

File: KALTURAPlayerSDK/KPAssetBuilder.m (hunk 5 above)
Comment (noamtamim) [inline KALTURAPlayerSDK/KPAssetBuilder.m]: I don't think it's a correct use of category.

**Pull request merged**